ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -96,20 +97,19 @@ func getClusterType(logger logr.Logger) bool {
 
 // NewReconciler returns a new reconcile.Reconciler
 func NewReconciler(mgr manager.Manager, options RayClusterReconcilerOptions) *RayClusterReconciler {
-	if err := mgr.GetFieldIndexer().IndexField(context.Background(), &corev1.Pod{}, podUIDIndexField, func(rawObj client.Object) []string {
+	ctx := context.Background()
+	if err := mgr.GetFieldIndexer().IndexField(ctx, &corev1.Pod{}, podUIDIndexField, func(rawObj client.Object) []string {
 		pod := rawObj.(*corev1.Pod)
 		return []string{string(pod.UID)}
 	}); err != nil {
 		panic(err)
 	}
-	log := ctrl.Log.WithName("controllers").WithName("RayCluster")
-	log.Info("Starting Reconciler")
-	isOpenShift := getClusterType(log)
+	isOpenShift := getClusterType(ctx)
 
 	return &RayClusterReconciler{
 		Client:            mgr.GetClient(),
 		Scheme:            mgr.GetScheme(),
-		Log:               log,
+		Log:               ctrl.Log.WithName("controllers").WithName("RayCluster"),
```

Comment:
An alternative would be to use the `LogConstructor` option, that can be set by calling `WithLogConstructor` with the controller builder API.

With that approach, the logger is shared between controller-runtime and the controller itself, and individual reconcile request can be traced with the `reconcileID` parameter.

Then the logger can be retrieved with `ctrl.LoggerFrom(ctx)` and is already fully set.

Reply:
Here is an example:
```go
// SetupWithManager builds the reconciler.
func (r *RayClusterReconciler) SetupWithManager(mgr ctrl.Manager, reconcileConcurrency int) error {
	b := ctrl.NewControllerManagedBy(mgr).
		For(&rayv1.RayCluster{}, builder.WithPredicates(predicate.Or(
			predicate.GenerationChangedPredicate{},
			predicate.LabelChangedPredicate{},
			predicate.AnnotationChangedPredicate{},
		))).
		Owns(&corev1.Pod{}).
		Owns(&corev1.Service{}).
		WithLogConstructor(func(request *reconcile.Request) logr.Logger {
			return ctrl.Log.WithName("controllers").
				WithName("RayCluster").
				WithValues("RayCluster", request.NamespacedName)
		})

	if EnableBatchScheduler {
		b = batchscheduler.ConfigureReconciler(b)
	}

	return b.
		WithOptions(controller.Options{MaxConcurrentReconciles: reconcileConcurrency}).
		Complete(r)
}
```

Reply:
Then we could get rid of the `r.Log` field.

Reply:
I will not spend time solving the issue in this PR. We can discuss whether to adopt this solution in a follow up PR.

Reply:
Open an issue to track the progress.